### PR TITLE
material-symbols: unstable-2023-11-17 -> 4.0.0-unstable-2024-05-17

### LINKED ISSUE #312230: Update request: material-symbols OLDVERSION → NEWVERSION
- Package name:material-symbols
- Latest released version: 2024.05.04

- Current version on the unstable channel: 2023.11.17
- Current version on the stable/release channel: 2023.11.17

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

**Notify maintainers**

-----

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by luftmensch-luftmensch**
## Description of changes
Closes  #312230

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by luftmensch-luftmensch** [comment]
```
> nix build .#material-symbols
> tree result/
 result
└──  share
   └──  fonts
      ├──  TTF
      │  ├──  MaterialSymbolsOutlined.ttf
      │  ├──  MaterialSymbolsRounded.ttf
      │  └──  MaterialSymbolsSharp.ttf
      └──  woff2
         ├──  MaterialSymbolsOutlined.woff2
         ├──  MaterialSymbolsRounded.woff2
         └──  MaterialSymbolsSharp.woff2
```

**Review by fufexan — APPROVED** [review]
Apart from the extra formatting of the argset, LGTM.

**Comment by luftmensch-luftmensch** [comment]
> Apart from the extra formatting of the argset, LGTM.

It was done by running `nixfmt-rfc`

**Comment by fufexan** [comment]
Okay then, no objections.

**Review by JohnRTitor — CHANGES_REQUESTED** [review]
Add the unstableGitUpdater update script in `passthru.updateScript` like in https://github.com/NixOS/nixpkgs/commit/f4c64495c45df2df14e2b195e8e737d50a3462a2

```nix
passthru.updateScript = unstableGitUpdater { };
```

**Comment by luftmensch-luftmensch** [comment]
> Add the unstableGitUpdater update script in `passthru.updateScript` like in [f4c6449](https://github.com/NixOS/nixpkgs/commit/f4c64495c45df2df14e2b195e8e737d50a3462a2)
> 
> ```nix
> passthru.updateScript = unstableGitUpdater { };
> ```

Done!

File: pkgs/data/fonts/material-symbols/default.nix [hunk 1]
```diff
@@ -28,12 +29,17 @@ stdenvNoCC.mkDerivation {
     runHook postInstall
   '';
 
+  passthru.updateScript = unstableGitUpdater { };
+
   meta = with lib; {
```
Comment (JohnRTitor) [inline pkgs/data/fonts/material-symbols/default.nix:34]: Remove `with lib;` from here, and add `lib.` to expressions that need it in the subset.

**Comment by luftmensch-luftmensch** [comment]
@JohnRTitor Should be good to go

**Review by JohnRTitor — APPROVED** [review]
LGTM

File: pkgs/data/fonts/material-symbols/default.nix [hunk 2]
```diff
@@ -1,18 +1,19 @@
...
+  stdenvNoCC,
+  fetchFromGitHub,
+  rename,
+  unstableGitUpdater,
 }:
-
 stdenvNoCC.mkDerivation {
   pname = "material-symbols";
-  version = "unstable-2023-11-17";
+  version = "unstable-2024-05-16";
```
[the remark this answers is not in the document]
Comment (luftmensch-luftmensch) [reply pkgs/data/fonts/material-symbols/default.nix]: Updated version to reflect the package naming

**Comment by luftmensch-luftmensch** [comment]
@OPNA2608 Should be resolved now

File: pkgs/data/fonts/material-symbols/default.nix (hunk 2 above)
[the remark this answers is not in the document]
Comment (JohnRTitor) [reply pkgs/data/fonts/material-symbols/default.nix]: The commit message and PR title should be updated as well.

[the remark this answers is not in the document]
Comment (luftmensch-luftmensch) [reply pkgs/data/fonts/material-symbols/default.nix]: There are both updated. Am I missing something?

**Comment by OPNA2608** [comment]
aarch64 build failed, looked like a temporary issue?

```
remote: Compressing objects: 100% (43/43), done.
error: RPC failed; curl 92 HTTP/2 stream 5 was not closed cleanly: CANCEL (err 8)
error: 4962 bytes of body are still expected
fetch-pack: unexpected disconnect while reading sideband packet
fatal: early EOF
fatal: fetch-pack: invalid index-pack output
Unable to checkout ace1af08508a6562ef05055a606cf44ea583ce3b from https://github.com/google/material-design-icons.git.
```

@ofborg build material-symbols material-symbols.passthru.tests

**Review by RossComputerGuy — APPROVED** [review]

**Review by OPNA2608 — APPROVED** [review]
LGTM.

**Pull request merged**